Documentation/kubernetes/install/kubespray.rst

```diff
@@ -10,7 +10,171 @@
 Installation using Kubespray
 ****************************
 
-Cilium is supported in the kubespray installer. Just follow the standard guide:
+The guide is to use Kubespray for creating an AWS Kubernetes cluster running 
+Cilium as the CNI. The **minimum** version required for this guide are:
 
-* `Deploy a Production Ready Kubernetes Cluster <https://github.com/kubernetes-incubator/kubespray/blob/master/README.md>`_
+  - Kubespray ``v2.6.0`` which uses Cilium ``v1.2.0`` as default
+ 
+Please consult `Kubespray Prerequisites <https://github.com/kubernetes-incubator/kubespray#requirements>`__ and Cilium :ref:`admin_system_reqs`. 
+
+
+Installing Kubespray
+====================
+
+.. code:: bash
+
+  $ git clone git@github.com:kubernetes-incubator/kubespray.git
+
+Install dependencies from ``requirements.txt``
+
+.. code:: bash
+
+  $ cd kubespray
+  $ sudo pip install -r requirements.txt
+
+
+Infrastructure Provisioning
+===========================
+
+We will use Terraform for provisioning AWS infrastructure.
+
+-------------------------
+Configure AWS credentials
+-------------------------
+
+Export the variables for your AWS credentials 
+
+.. code:: bash
+
+  export AWS_ACCESS_KEY_ID="www"
+  export AWS_SECRET_ACCESS_KEY ="xxx"
+  export AWS_SSH_KEY_NAME="yyy"
+  export AWS_DEFAULT_REGION="zzz"
+
+-----------------------------
+Configure Terraform Variables
+-----------------------------
+
+Navigate to the kubespray cloned repo directory and rename 
+``contrib/terraform/aws/terraform.tfvars.example`` to ``terraform.tfvars``. Open the 
+file and change any defaults particularly the number of master, etcd, and worker nodes. 
+You can change the master and etcd number to 1 if using the cluster just for demo and 
```

Comment:
You can change the number of master and etcd nodes to 1 if you are using the cluster only for demoing or testing purposes.